onnxruntime/core/providers/cuda/generator/range_impl.cu

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <cub/cub.cuh>
+#include <cublas_v2.h>
+#include <cuda_fp16.h>
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "core/providers/cuda/cuda_common.h"
+#include "range_impl.h"
+
+using namespace onnxruntime::cuda;
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T, unsigned TPB>
+__global__ void RangeKernel(const T start, const T delta, const size_t n, T* output) {
+  int index = blockIdx.x * blockDim.x + threadIdx.x;
+  if (index < n) {
+      T value = start + delta * index;
+      T value_delta = delta * TPB;
+      for (int i = index; i < n; i += TPB, value += value_delta) {
```

Comment:
>for (int i = index; i < n [](start = 6, length = 25)

seems not necessary? from the logic, it seems the same threadIdx in different blocks would write to the same address with the same value.

Reply:
@KeDengMS, Good catch! I've updated the code. Please take a look.